airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -0,0 +1,422 @@
+        for resp in self.read_records(SyncMode.full_refresh):
+            return resp, None
+        return None, "not found settings"
+
+
+class IncrementalBasicSearchStream(SourceZendeskSupportStream, ABC):
+    """Base class for all data lists with a incremental stream"""
+
+    # max size of one data chunk. 100 is limitation of ZenDesk
+    state_checkpoint_interval = SourceZendeskSupportStream.page_size
+
+    # default sorted field
+    cursor_field = SourceZendeskSupportStream.updated_at_field
+
+    def __init__(self, start_date: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # add the custom value for skiping of not relevant records
+        self._start_date = self.str2datetime(start_date) if isinstance(start_date, str) else start_date
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        next_page = self._parse_next_page_number(response)
+        if next_page:
+            return {"next_page": next_page}
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        # Root of all responses of searching endpoints is 'results'
+        yield from response.json()["results"] or []
+
+    def path(self, *args, **kargs) -> str:
+        return "search.json"
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        updated_after = None
+        if stream_state and stream_state.get(self.cursor_field):
+            updated_after = self.str2datetime(stream_state[self.cursor_field])
+
+        # add the 'query' parameter
+        conds = [f"type:{self.entity_type[:-1]}"]
+        conds.append("created>%s" % self.datetime2str(self._start_date))
+        if updated_after:
+            conds.append("updated>%s" % self.datetime2str(updated_after))
+
+        res = {
+            "query": " ".join(conds),
+            "sort_by": self.updated_at_field,
+            "sort_order": "desc",
+            "size": self.state_checkpoint_interval,
+        }
+        if next_page_token:
+            res["page"] = next_page_token["next_page"]
+        return res
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        # try to save maximum value of a cursor field
+        return {
+            self.cursor_field: max(
+                (latest_record or {}).get(self.cursor_field, ""), (current_stream_state or {}).get(self.cursor_field, "")
+            )
+        }
+
+
+class IncrementalBasicEntityStream(IncrementalBasicSearchStream, ABC):
+    """basic stream for endpoints where an entity name can be used in a path value
+    https://<subdomain>.zendesk.com/api/v2/<entity_name>.json
+    """
+
+    # for generation of a path value and as rule as JSON root name of all response
+    entity_type: str = None
```

Comment:
is it stream_name ? if yes - we can just use self.name

Reply:
as rule as it is same but some streams have custom values